app/models/attestation_template.rb

```diff
@@ -43,17 +47,68 @@ def procedure_type_de_champ_tags
   end
 
   def individual_tags
-    [{ libelle: 'civilité', description: 'M., Mme' },
-     { libelle: 'nom', description: "nom de l'usager" },
-     { libelle: 'prénom', description: "prénom de l'usager" }]
+    [{ libelle: 'civilité', description: 'M., Mme', target: 'gender' },
+     { libelle: 'nom', description: "nom de l'usager", target: 'nom' },
+     { libelle: 'prénom', description: "prénom de l'usager", target: 'prenom' }]
   end
 
   def entreprise_tags
-    [{ libelle: 'SIREN', description: '' },
-     { libelle: 'numero de TVA intracommunautaire', description: '' },
-     { libelle: 'SIRET du siège social', description: '' },
-     { libelle: 'nom commmercial', description: '' },
-     { libelle: 'raison social', description: '' }]
+    [{ libelle: 'SIREN', description: '', target: 'siren' },
+     { libelle: 'numero de TVA intracommunautaire', description: '', target: 'numero_tva_intracommunautaire' },
+     { libelle: 'SIRET du siège social', description: '', target: 'siret_siege_social' },
+     { libelle: 'raison social', description: '', target: 'raison_sociale' }]
+  end
+
+  def build_pdf(dossier)
+    action_view = ActionView::Base.new(ActionController::Base.view_paths,
+                                       logo: logo,
+                                       title: replace_tags(title, dossier),
+                                       body: replace_tags(body, dossier),
+                                       signature: signature,
+                                       footer: footer,
+                                       created_at: Time.now)
+
+    attestation_view = action_view.render(file: 'admin/attestation_templates/show',
+                                          formats: [:pdf])
+
+    view_to_memory_file(attestation_view)
+  end
+
+  def view_to_memory_file(view)
+    pdf = StringIO.new(view)
+    def pdf.original_filename; 'attestation'; end
```

Comment:
Voir https://github.com/carrierwaveuploader/carrierwave/wiki/How-to:-Upload-from-a-string-in-Rails-3
L'autre solution est de créer une classe spécifique héritant de StringIO et de lui rajouter la méthode original_filename.

Tu préfères ?